sklearn/ensemble/_hist_gradient_boosting/histogram.pyx

```diff
@@ -153,11 +166,18 @@ cdef class HistogramBuilder:
                         ordered_gradients[i] = gradients[sample_indices[i]]
                         ordered_hessians[i] = hessians[sample_indices[i]]
 
-            for feature_idx in prange(n_features, schedule='static',
-                                      num_threads=n_threads):
-                # Compute histogram of each feature
+            # Compute histogram of each feature
+            for f_idx in prange(
+                n_allowed_features, schedule='static', num_threads=n_threads
+            ):
+                if has_interaction_cst:
+                    feature_idx = allowed_features[f_idx]
+                else:
+                    feature_idx = f_idx
```

Comment:
Are there performances benefit of making the branching outside the `prange` loops and have one `prange` loop per branch?

Reply:
I tried that, but I did not observe any runtime difference. Usually, the number of features is small (less than 10, 100, maybe 1000), but sample size is large. The inner loop is then over n_samples and there branching should be avoided.